part-1.py

```diff
@@ -5,16 +5,52 @@
 
 # factorial
 
-
+def factorial(n):
+    if n == 0:
+        return 1
+    elif n > 0: 
+        return n * (factorial(n-1))
+    else:
+        raise ValueError
 
 # reverse
 
-
+def reverse(text):
```

Comment:
👍 Just a note that `text[1:]`  makes a copy of the array element-by-element so that makes this function O(n^2) in time/space complexity.